### ISSUE TITLE
Package Photoprism

### ISSUE THREAD

**Issue by danielfullmer** [NixOS/nixpkgs#96043]
**Project description**
PhotoPrism is a server-based application for browsing, organizing and sharing your personal photo collection.

**Metadata**

* homepage URL: https://photoprism.org/
* source URL: https://github.com/photoprism/photoprism
* license: agpl3
* platforms: linux, darwin, windows

**Comment by Sohalt** [comment]
I've started a PR at https://github.com/NixOS/nixpkgs/pull/99016. It's basically working, but I would love to have some feedback on some unresolved questions.

**Comment by ajs124** [comment]
There's also https://github.com/andir/infra/tree/master/nix/packages/photoprism

**Comment by Sohalt** [comment]
Welp. Didn't see that :/

**Comment by mbrgm** [comment]
Still important. @Sohalt any news on the PR?

**Comment by Sohalt** [comment]
@mbrgm I'm still waiting on feedback on https://github.com/NixOS/nixpkgs/pull/99016#discussion_r531681435 and I think it would be cleaner to use npmlock2nix instead of node2nix, but I didn't find the time to get that working.

**Comment by newAM** [comment]
I have this packaged at the 2022-01-21 release based on all the prior work:
* [package](https://github.com/newAM/nixpkgs/tree/photoprism/pkgs/servers/web-apps/photoprism)
* [module](https://github.com/newAM/nixpkgs/blob/photoprism/nixos/modules/services/web-apps/photoprism.nix)

I will pull-request that in a few months if I end up using photoprism long term.
If someone wants to maintain photoprism before then feel free to take over.

**Comment by Misterio77** [comment]
I was hoping to run it on my Pi, so here's a package that supports both aarch64 and x86_64 (using photoprism's prebuilt tensorflow .so's): https://github.com/Misterio77/nix-config/tree/c6c37ec16711b6a717587d02c3eb0627aa388c70/pkgs/photoprism

**Comment by Stunkymonkey** [comment]
i would be very happy to get this package included in 22.05. Maybe @newAM you could open a PR?

**Comment by newAM** [comment]
I have not been using photoprism much unfortunately, I am not sure when I will get back to it.

**Comment by Misterio77** [comment]
I've been using photoprism daily, thus am interested in upstreaming and maintaining it, if that would be okay :)

**Comment by newAM** [comment]
Take whatever you need from what I have written!

**Comment by newAM** [comment]
@Misterio77 opened a PR here: https://github.com/NixOS/nixpkgs/pull/170953
(Mentions do not generate notifications; people subscribed to this issue probably want to review the PR)

**Comment by skogsbrus** [comment]
Following this with great interest, let me know if I can help!

**Comment by BeneSim** [comment]
@Misterio77 Unfortunately the latest [commit](https://github.com/NixOS/nixpkgs/pull/170953/commits/7757aef5b01cf01c3e6abbaaf3af134f18c02e18) has rendered the PR unbuildable for me.

I branched of your PR, fixed the issues introduced by the commit, updated to the photoprism version that has been released 2 days ago (221102-905925b4d ) and rebased the PR on a more recent  rev of master (was needed  anyway since the new photoprism version required go 1.19). I also implemented a couple of suggestions by @dotlambda left over in the PR.

https://github.com/NixOS/nixpkgs/compare/master...BeneSim:nixpkgs:photoprism

I'd be willing to take over the PR, you mentioned that you didn't end up using photprism as much. Would be my first contribution to nixpkgs, but I'm interested and willing to contribute ;)

Let me know if that would be okay for you.

Until then, anybody who wants to use  photoprism today with nixpkgs might want to have a look at my branch https://github.com/BeneSim/nixpkgs/tree/photoprism .
